install 4.1.1 on M2 chip failed

`pip install 4.0.1` is ok, which seems to just download binary code and install. However, source code compilation is required for 4.1.1. I got an error in rust(version = 1.74.0):
```
       Running `/Users/dp/.rustup/toolchains/stable-aarch64-apple-darwin/bin/rustc --crate-name build_script_build --edition=2021 /Users/dp/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.70/build.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debug-assertions=off -C overflow-checks=on --cfg 'feature="proc-macro"' -C metadata=ffe8ac52cd8a21e2 -C extra-filename=-ffe8ac52cd8a21e2 --out-dir /private/var/folders/2c/h7ddyfg16y57l8ddqyqt171h0000gn/T/pip-install-szjre90g/bcrypt_020353eb77d844e0bf2265266882c383/src/_bcrypt/target/release/build/proc-macro2-ffe8ac52cd8a21e2 -L dependency=/private/var/folders/2c/h7ddyfg16y57l8ddqyqt171h0000gn/T/pip-install-szjre90g/bcrypt_020353eb77d844e0bf2265266882c383/src/_bcrypt/target/release/deps --cap-lints allow`
  error[E0463]: can't find crate for `core`
    |
    = note: the `x86_64-apple-darwin` target may not be installed
    = help: consider downloading the target with `rustup target add x86_64-apple-darwin`
  
  error[E0463]: can't find crate for `compiler_builtins`
```
I don't think I should install `x86_64-apple-darwin` on macos with an M2 chip.

--- a/.github/workflows/wheel-builder.yml
+++ b/.github/workflows/wheel-builder.yml
@@ -155,7 +155,7 @@ jobs:
           venv/bin/python -m pip wheel -v $PY_LIMITED_API bcrypt*.tar.gz -w dist/ && mv dist/bcrypt*.whl wheelhouse
         env:
           PYTHON_VERSION: ${{ matrix.PYTHON.ABI_VERSION }}
-          DEPLOYMENT_TARGET: '10.12'
+          MACOSX_DEPLOYMENT_TARGET: '10.12'
           ARCHFLAGS: '-arch arm64 -arch x86_64'
           _PYTHON_HOST_PLATFORM: 'macosx-10.9-universal2'
       - run: venv/bin/pip install -f wheelhouse --no-index bcrypt

